### ISSUE TITLE
Add content labels

### ISSUE THREAD

**Issue by magi42** [vaadin/docs#492]
Add content labels for AsciiDoc, Java, and TypeScript when a pull is created, as well as corresponding "AsciiDoc OK", and so on, labels for each content type. Once a review is finished the reviewer should change the label to "<type> OK". Once all are OK, the PR can be merged at any time.

Suggested choices for the labeling actions are:

* https://github.com/actions/labeler
* https://github.com/marketplace/actions/auto-label-action

Considerations:

* Files can be added or removed from a PR after creating
* Files can be edited after accepting a PR (turning one of the labels to OK
* Merging main branch triggers labeling check, so if the labeler acts on that and re-assigns the original labels even when nothing changed, that would be bad. So we probably need to keep further label updates manual.
* If based solely on file types, doesn't consider inline code snippets in AsciiDoc
* Some changes are only in code snippets, so AsciiDoc label would be wrong
* Other content such as CSS is difficult to consider

Optional automatic merging once all labels are "OK":

* https://stackoverflow.com/questions/59588605/how-to-check-for-a-label-in-a-github-action-condition
* https://github.com/marketplace/actions/verify-pull-request-labels
* https://github.com/marketplace/actions/require-labels

Tasks:

- [x] Add content labels and corresponding OK labels
- [x] Add labels automatically on new PR
- [x] Document the procedure and guidelines in Contributing to Docs

**Comment by jouni** [comment]
This process is now in place.
